README.md
This project is a fork of [FRESteamWorks by Ventero](https://github.com/Ventero/FRESteamWorks/)
to adapt the Steam API changes in v1.3: Workshop access in ISteamRemoteStorage had been replaced by ISteamUGC.

Methods added:
```
createQueryUserUGCRequest(userId:String, listType:uint, matchingUGCType:uint, sortOrder:uint, creatorAppId:uint, consumerAppId:uint, page:uint):String
createQueryAllUGCRequest(queryType:uint, matchingUGCType:uint, creatorAppId:uint, consumerAppId:uint, page:uint):String
createQueryUGCDetailsRequest(publishedFileIds:Array):String
sendQueryUGCRequest(handle:String):Boolean
getQueryUGCResult():Array
createItem(consumerAppId:uint, fileType:uint):Boolean
createItemResult():String
startItemUpdate(consumerAppId:uint, fileId:String):String
setItemTitle(handle:String, title:String):Boolean
setItemDescription(handle:String, desc:String):Boolean
setItemUpdateLanguage(handle:String, lang:String):Boolean
setItemMetadata(handle:String, metadata:String):Boolean
setItemVisibility(handle:String, visibility:uint):Boolean
setItemTags(handle:String, tags:Array):Boolean
setItemContent(handle:String, contentFolder:String):Boolean
setItemPreview(handle:String, previewPath:String):Boolean
addItemKeyValueTag(handle:String, key:String, value:String):Boolean
removeItemKeyValueTags(handle:String, key:String):Boolean
submitItemUpdate(handle:String, changeNote:String):Boolean
subscribeItem(publishedFileId:String):Boolean
unsubscribeItem(publishedFileId:String):Boolean
getSubscribedItems():Array
getItemState(publishedFileId:String):uint
getItemInstallInfo(publishedFileId:String):UGCItemInstallInfo
getItemDownloadInfo(publishedFileId:String):Array
downloadItem(publishedFileId:String, highPriority:Boolean):Boolean
```

Methods removed:
```
UGCDownload(handle:String, priority:uint):Boolean
UGCRead(handle:String, size:int, offset:uint, data:ByteArray):Boolean
getUGCDownloadProgress(handle:String):Array
getUGCDownloadResult(handle:String):DownloadUGCResult
publishWorkshopFile(name:String, preview:String, appId:uint, title:String, description:String, visibility:uint, tags:Array, fileType:uint):Boolean
publishWorkshopFileResult():String
deletePublishedFile(file:String):Boolean
getPublishedFileDetails(file:String, maxAge:uint = 0):Boolean
getPublishedFileDetailsResult(file:String):FileDetailsResult
enumerateUserPublishedFiles(startIndex:uint):Boolean
enumerateUserPublishedFilesResult():UserFilesResult
enumeratePublishedWorkshopFiles(type:uint, start:uint, count:uint, days:uint, tags:Array, userTags:Array):Boolean
enumeratePublishedWorkshopFilesResult():WorkshopFilesResult
enumerateUserSubscribedFiles(startIndex:uint):Boolean
enumerateUserSubscribedFilesResult():SubscribedFilesResult
enumerateUserSharedWorkshopFiles(steamID:String, start:uint, required:Array, excluded:Array):Boolean
enumerateUserSharedWorkshopFilesResult():UserFilesResult
enumeratePublishedFilesByUserAction(action:uint, startIndex:uint):Boolean
enumeratePublishedFilesByUserActionResult():FilesByUserActionResult
subscribePublishedFile(file:String):Boolean
unsubscribePublishedFile(file:String):Boolean
createPublishedFileUpdateRequest(file:String):String
updatePublishedFileFile(handle:String, file:String):Boolean
updatePublishedFilePreviewFile(handle:String, preview:String):Boolean
updatePublishedFileTitle(handle:String, title:String):Boolean
updatePublishedFileDescription(handle:String, description:String):Boolean
updatePublishedFileSetChangeDescription(handle:String, changeDesc:String):Boolean
updatePublishedFileVisibility(handle:String, visibility:uint):Boolean
updatePublishedFileTags(handle:String, tags:Array):Boolean
commitPublishedFileUpdate(handle:String):Boolean
getPublishedItemVoteDetails(file:String):Boolean
getPublishedItemVoteDetailsResult():ItemVoteDetailsResult
getUserPublishedItemVoteDetails(file:String):Boolean
getUserPublishedItemVoteDetailsResult():UserVoteDetails
updateUserPublishedItemVote(file:String, upvote:Boolean):Boolean
setUserPublishedFileAction(file:String, action:uint):Boolean
```

# Steamworks API for AIR applications #

A simple Steamworks API wrapper to be used as native extension for Adobe AIR
applications. This allows you to implement Steam achievements, leaderboards, cloud
storage, workshop support and much more in your AIR application - on Windows,
OS X and Linux.

This project initially started as a fork of [FRESteamWorks by Oldes/Amanita Design](https://github.com/Oldes/FRESteamWorks/).

If you want to contribute to this project, please take a look at the
[contributing guide](https://github.com/indiest/FRESteamWorks/tree/master/CONTRIBUTING.md).

For a list of changes, see [the changelog](https://github.com/indiest/FRESteamWorks/tree/master/CHANGELOG.md).

# Download #

Pre-built cross-platform ANEs (for both Windows and OS X), as well as Linux
SWCs can be found on http://dump.ventero.de/FRESteamWorks/. The Linux wrapper
binary has to be built manually, see [the Linux section](#linux) for more details.

# API Documentation #

For a full list of all supported functions with some accompanying documentation,
see [lib/API.txt](https://github.com/indiest/FRESteamWorks/blob/master/lib/API.txt).
In general, the FRESteamWorks API functions try to be a close representation of
the native Steamworks SDK functions. Thus, for detailed documentation, see the
Steamworks SDK docs.

The objects returned by certain API functions are plain data structures.
For a list of available properties, see [the corresponding source files](https://github.com/indiest/FRESteamWorks/tree/master/lib/src/com/amanitadesign/steam).

# Usage #

For Windows/OS X builds, you'll only have to include the ANE in your project like
any normal SWC, add the extension to your application descriptor and include
`extendedDesktop` in the list of supported profiles. For an example, see
[the application descriptor of the FRESteamWorks test application](https://github.com/indiest/FRESteamWorks/blob/master/test/bin-debug/FRESteamWorksTest-app.xml#L14-17).

Testing your application is a bit more complicated and depends on how your project
is built. For detailed information, see below. For general information regarding
the testing of Steam applications, please refer to the Steamworks SDK documentation.

Since the AIR runtime on Linux doesn't support native extensions, an external
binary is used to communicate with the Steamworks API. For more information
on how to build and include this tool, see [the Linux section](#linux).

## Known Issues ##

Due to a bug in old versions of the AIR runtime, the runtime will crash as soon
as a function is called that returns any of the `com.amanitadesign.steam.*` objects.
For this reason, FRESteamWorks should only be used with AIR runtimes >= 3.7.

Flash Builder 4.6 seems to have an issue with detecting classes that implement
interfaces inside an ANE, which results in the `FRESteamWorks` class not being
visible inside the ANE within Flash Builder when using earlier builds (older than
v0.4-19) of the ANE. This issue is also fixed in Flash Builder 4.7.

With older versions of the ANE (older than v0.4-19), running an application
including that ANE is not possible in Flash Builder 4.6 on OS X. To fix this issue,
simply download to a more recent ANE.

## Flash Builder ##

### Building ###

To include the ANE in your Flash Builder project, simply add it to the list of
"Native Extensions" under "ActionScript Build Path" in your project's properties.
Make sure it gets packaged when building your application by checking the "Package" box
in the "Native Extensions" tab under "ActionScript Build Packaging". The extension
will automatically be added to your application descriptor, but you still might have
to add `extendedDesktop` to the list of supported profiles.

### Running ###

Since by default, Flash Builder uses its installation path as working directory
when running applications, this however isn't enough to test your application in
Flash Builder. This can be fixed by either adding the library's directory to the
dynamic linker's search path (see [the AIR/Flex SDK section](#airflex-sdk) on
how to do this), or by creating a new shortcut to Flash Builder where the working
directory is set to your project's directory and adding the dynamic library to
your project (this does not work on OS X).

### Packaging ###

Just include the native Steamworks library at the top level of your project so
that it gets included when packaging your application.

## AIR/Flex SDK ##

### Building ###

For projects built with the AIR/Flex SDKs, the ANE can be included by simply
adding the ANE path to the external library path (either by adding it to your
build config file, or by adding `-external-library-path+=/path/to/FRESteamWorks.ane`
to your `mxmlc`/`compc` compiler flags). The SWF you're has to be at least version 11.

### Running ###

To run your application with the AIR Debug Launcher (`adl`), you will have to first
unpack the ANE. To do this, rename `FRESteamWorks.ane` to `FRESteamWorks.zip`,
then unzip it and rename the generated folder to `FRESteamWorks.Unpacked.ane`.
Now you can add `-extdir /path/to/folder` to the `adl` command line flags. Here,
`folder` refers the parent folder of the unzipped extension (i.e. the folder that
contains `FRESteamWorks.Unpacked.ane`).

Since the native library included in the ANE is dynamically linked against the
Steamworks library, you'll also have to make sure that the dynamic linker can find
the Steamworks library. When testing, this is easiest done by adding the folder containing
the Steamworks library to your `%PATH%` environment variable on Windows
([example](https://github.com/indiest/FRESteamWorks/blob/master/test/bin-debug/runWin.bat#L9)),
`DYLD_FALLBACK_LIBRARY_PATH` on OS X ([example](https://github.com/indiest/FRESteamWorks/blob/master/test/bin-debug/runMac.sh#L6))
or the library itself to `LD_PRELOAD` on Linux ([example](https://github.com/indiest/FRESteamWorks/blob/master/test/bin-debug/runLinux.sh#L24)).

### Packaging ###

For packaged builds, you'll simply have to include the native Steamworks libraries
in the top level of your build and add the path to a directory containing the
`FRESteamWorks.ane` as `-extdir` to `adt`. For an example, see the test application's
[packaging script](https://github.com/indiest/FRESteamWorks/blob/master/test/bin-debug/packageWin.bat#L13-18).

## Flash Professional ##

### Building ###

First of all, make sure that your project's target is set to "AIR 3.0 for Desktop"
(or any more recent AIR version). Then, in the AIR preferences (wrench icon next
to target project's target), check the "Extended Desktop" profile and select
"Output as: Application with runtime embedded".

To include the ANE, go to the "Advanced ActionScript 3.0 Settings" (wrench icon
next to the project's script settings), and in the "Library path" tab,
click the native extension button and select `FRESteamWorks.ane`.

### Running ###

See the [Flash Builder section](#running)
on running the application.

### Packaging ###

See the [Flash Builder section](#packaging)
on packaging the application.

## FDT ##

FDT is very similar to [Flash Builder](#flash-builder). The main differences
are that the ANE is automatically added to the project and packaged when copying
it into the project's `lib` folder, so that the steps described in
[Flash Builder's building section](#building) are not necessary. However, when adding
the native Steamworks libraries, you have to make sure that they're packaged with
the application by opening the project's properties and adding them under
"FDT AIR Properties" -> "Desktop" -> "Package Contents".

Please also make sure to follow the steps described in [the running section](#running)
to debug your application from within FDT and see the [list of known issues](#known-issues).

# Linux #

The AIR runtime on Linux doesn't support native extensions. Instead, you'll have
to compile your AS3 application against `FRESteamWorksLibLinux.swc` and compile
a wrapper binary that handles the communication between AIR and the Steamworks API.
For details on how to do that, please contact me directly (email address see profile).

# Building FRESteamWorks #

There shouldn't be any reason for you to manually build the FRESteamWorks.ane,
as pre-built ANEs can be downloaded from http://dump.ventero.de/FRESteamWorks/.
If you need a more recent version than the builds that are available on that site,
you can create an issue in the [bug tracker](https://github.com/indiest/FRESteamWorks/issues).

If you still want to build the ANE yourself (or build the test application), a few
simple steps have to be followed.

1. Create a config file by copying `config_example.sh` to `config.sh` on OS X,
or `config_example.bat` to `config.bat` and correctly setting up the values within.

2. Open the Visual Studio solution or XCode project (both inside `src/`) and fix
up the include paths (for the Adobe AIR SDK as well as the Steamworks SDK).

3. Build the project in either debug or release mode. This compiles the native
library and automatically builds an ANE for the current platform containing that
library.

4. Optionally, build the test application by running `build.{bat,sh}` inside
`test/bin-debug/` and run it with `runMac.sh` or `runWin.bat`. This automatically
uses the ANE built in step 3.

To build a cross platform ANE, you'll have to compile the native library on both
Windows and OS X and then create an ANE that includes both of those. This is easiest
done by first running `mkdir.sh` in `builds/`, building FRESteamWorks.dll on Windows,
copying it over to the directory created on OS X and then running `compile.sh`
and `build.sh` in that order.

For more details, please see the
[contributing guide](https://github.com/indiest/FRESteamWorks/tree/master/CONTRIBUTING.md).

---

# License #

See [LICENSE](https://github.com/indiest/FRESteamWorks/tree/master/LICENSE).


CHANGELOG.md
Changelog
=========

The builds described in this file can be [downloaded here](http://dump.ventero.de/FRESteamWorks/).
A full list of implemented functions can be found by following the link in the
version number.

### [HEAD](https://github.com/indiest/FRESteamWorks/blob/HEAD/lib/API.txt)

### [v0.6]
 - Removed all wrapper functions deprecated in ISteamRemoteStorage.
 - Added wrapper functions added in ISteamUGC.

### [v0.5-29-gea3356b](https://github.com/Ventero/FRESteamWorks/blob/v0.5-29-gea3356b/lib/API.txt)
 - Added `setEnv` to allow setting environment variables.
 - Fixed `enumerateUserPublishedFilesResult` to correctly return all results ([issue #13](https://github.com/Ventero/FRESteamWorks/issues/13)).
 - Implemented `indicateAchievementProgress`.
 - Removed the deprecated (and broken) `useCrashHandler`.

### [v0.5-2-gf59d5ba](https://github.com/Ventero/FRESteamWorks/blob/v0.5-2-gf59d5ba/lib/API.txt)
 - Fixed `microTxnResult` to properly return null if no results available, and correctly initialize the `authorized` property for returned objects.

### [v0.5](https://github.com/Ventero/FRESteamWorks/blob/v0.5/lib/API.txt)
 - Implemented `setOverlayNotificationPosition`
 - Added a callback for microtransaction authorizations (thanks to @Unsanctioned)
 - Added `restartAppIfNecessary` (again, thanks to @Unsanctioned)
 - Exposed `startProcess(path:String)` on Linux, which has to be called before any other API function if the API wrapper binary is in a non-standard location
 - Deprecated the (undocumented) `path` argument to `init` on Linux and `useCrashHandler`

### [v0.4-54-gdd1b399](https://github.com/Ventero/FRESteamWorks/blob/v0.4-54-gdd1b399/lib/API.txt)
 - Implemented global stats

### [v0.4-46-g8526828](https://github.com/Ventero/FRESteamWorks/blob/v0.4-46-g8526828/lib/API.txt)
 - Fixed `getStat{Int,Float}` on Linux (which would previously always return 0)
 - Added support for basic authentication & ownership checks

### [v0.4-36-g5ff1e97](https://github.com/Ventero/FRESteamWorks/blob/v0.4-36-g5ff1e97/lib/API.txt)
 - Added `findOrCreateLeaderboard`
 - Implemented basic friends API

### [v0.4-30-g207935d](https://github.com/Ventero/FRESteamWorks/blob/v0.4-30-g207935d/lib/API.txt)
 - Handle certain cases of API failures more gracefully
 - Implemented `getEnv`

### [v0.4-19-g0cc78a0](https://github.com/Ventero/FRESteamWorks/blob/v0.4-19-g0cc78a0/lib/API.txt)
 - Work around a bug in Flash Builder where certain classes from the ANE would be ignored.

### [v0.4](https://github.com/Ventero/FRESteamWorks/blob/v0.4/lib/API.txt)
 - Added Steam Workshop support
 - Various leaderboards related functions added
 - Added support for some DLC related functions
 - Implemented several `activateGameOverlayTo___` functions


src/CSteam/ResponseTypes.h
/*
 *  ResponseTypes.h
 *  This file is part of FRESteamWorks.
 *
 *  Created by Ventero <http://github.com/Ventero> on 2014-04-17.
 *  Copyright (c) 2014 Level Up Labs, LLC. All rights reserved.
 */

#ifndef RESPONSETYPES_H
#define RESPONSETYPES_H

// General result codes
// Remember to run lib/generateAPI.rb when changing this.
enum ResponseTypes {
	RESPONSE_OnUserStatsReceived,
	RESPONSE_OnUserStatsStored,
	RESPONSE_OnAchievementStored,
	RESPONSE_OnGlobalStatsReceived,
	RESPONSE_OnFindLeaderboard,
	RESPONSE_OnUploadLeaderboardScore,
	RESPONSE_OnDownloadLeaderboardEntries,
	RESPONSE_OnGameOverlayActivated,
	RESPONSE_OnFileShared,
	RESPONSE_OnUGCDownload,
	RESPONSE_OnPublishWorkshopFile,
	RESPONSE_OnDeletePublishedFile,
	RESPONSE_OnGetPublishedFileDetails,
	RESPONSE_OnEnumerateUserPublishedFiles,
	RESPONSE_OnEnumeratePublishedWorkshopFiles,
	RESPONSE_OnEnumerateUserSubscribedFiles,
	RESPONSE_OnEnumerateUserSharedWorkshopFiles,
	RESPONSE_OnEnumeratePublishedFilesByUserAction,
	RESPONSE_OnCommitPublishedFileUpdate,
	RESPONSE_OnSubscribePublishedFile,
	RESPONSE_OnUnsubscribePublishedFile,
	RESPONSE_OnGetPublishedItemVoteDetails,
	RESPONSE_OnGetUserPublishedItemVoteDetails,
	RESPONSE_OnUpdateUserPublishedItemVote,
	RESPONSE_OnSetUserPublishedFileAction,
	RESPONSE_OnGetAuthSessionTicketResponse,
	RESPONSE_OnValidateAuthTicketResponse,
	RESPONSE_OnDLCInstalled,
	RESPONSE_OnMicroTxnAuthorizationResponse,
	RESPONSE_OnCreateItem,
	RESPONSE_OnSubmitItemUpdate,
	RESPONSE_OnQueryUGC,
	RESPONSE_OnSubscribeItem,
	RESPONSE_OnUnsubscribeItem,
	RESPONSE_OnDownloadItem
};

#endif


src/functions.h
/* automatically generated by generateAPI.rb */
/* START GENERATED CODE */
/********************************/
/* General Steamworks functions */
/********************************/
X(AIRSteam_Init) /* = 0 */
X(AIRSteam_RunCallbacks) /* = 1 */
X(AIRSteam_GetUserID) /* = 2 */
X(AIRSteam_GetAppID) /* = 3 */
X(AIRSteam_GetAvailableGameLanguages) /* = 4 */
X(AIRSteam_GetCurrentGameLanguage) /* = 5 */
X(AIRSteam_GetPersonaName) /* = 6 */
X(AIRSteam_RestartAppIfNecessary) /* = 7 */
/************************/
/* Stats / Achievements */
/************************/
X(AIRSteam_RequestStats) /* = 8 */
X(AIRSteam_SetAchievement) /* = 9 */
X(AIRSteam_ClearAchievement) /* = 10 */
X(AIRSteam_IsAchievement) /* = 11 */
X(AIRSteam_IndicateAchievementProgress) /* = 12 */
X(AIRSteam_GetStatInt) /* = 13 */
X(AIRSteam_GetStatFloat) /* = 14 */
X(AIRSteam_SetStatInt) /* = 15 */
X(AIRSteam_SetStatFloat) /* = 16 */
X(AIRSteam_StoreStats) /* = 17 */
X(AIRSteam_ResetAllStats) /* = 18 */
X(AIRSteam_RequestGlobalStats) /* = 19 */
X(AIRSteam_GetGlobalStatInt) /* = 20 */
X(AIRSteam_GetGlobalStatFloat) /* = 21 */
X(AIRSteam_GetGlobalStatHistoryInt) /* = 22 */
X(AIRSteam_GetGlobalStatHistoryFloat) /* = 23 */
/****************/
/* Leaderboards */
/****************/
X(AIRSteam_FindLeaderboard) /* = 24 */
X(AIRSteam_FindOrCreateLeaderboard) /* = 25 */
X(AIRSteam_FindLeaderboardResult) /* = 26 */
X(AIRSteam_GetLeaderboardName) /* = 27 */
X(AIRSteam_GetLeaderboardEntryCount) /* = 28 */
X(AIRSteam_GetLeaderboardSortMethod) /* = 29 */
X(AIRSteam_GetLeaderboardDisplayType) /* = 30 */
X(AIRSteam_UploadLeaderboardScore) /* = 31 */
X(AIRSteam_UploadLeaderboardScoreResult) /* = 32 */
X(AIRSteam_DownloadLeaderboardEntries) /* = 33 */
X(AIRSteam_DownloadLeaderboardEntriesResult) /* = 34 */
/**************************/
/* Cloud / Remote Storage */
/**************************/
X(AIRSteam_GetFileCount) /* = 35 */
X(AIRSteam_GetFileSize) /* = 36 */
X(AIRSteam_FileExists) /* = 37 */
X(AIRSteam_FileWrite) /* = 38 */
X(AIRSteam_FileRead) /* = 39 */
X(AIRSteam_FileDelete) /* = 40 */
X(AIRSteam_FileShare) /* = 41 */
X(AIRSteam_FileShareResult) /* = 42 */
X(AIRSteam_IsCloudEnabledForApp) /* = 43 */
X(AIRSteam_SetCloudEnabledForApp) /* = 44 */
X(AIRSteam_GetQuota) /* = 45 */
/******************/
/* UGC / Workshop */
/******************/
X(AIRSteam_CreateQueryUserUGCRequest) /* = 46 */
X(AIRSteam_CreateQueryAllUGCRequest) /* = 47 */
X(AIRSteam_CreateQueryUGCDetailsRequest) /* = 48 */
X(AIRSteam_SendQueryUGCRequest) /* = 49 */
X(AIRSteam_GetQueryUGCResult) /* = 50 */
X(AIRSteam_CreateItem) /* = 51 */
X(AIRSteam_CreateItemResult) /* = 52 */
X(AIRSteam_StartItemUpdate) /* = 53 */
X(AIRSteam_SetItemTitle) /* = 54 */
X(AIRSteam_SetItemDescription) /* = 55 */
X(AIRSteam_SetItemUpdateLanguage) /* = 56 */
X(AIRSteam_SetItemMetadata) /* = 57 */
X(AIRSteam_SetItemVisibility) /* = 58 */
X(AIRSteam_SetItemTags) /* = 59 */
X(AIRSteam_SetItemContent) /* = 60 */
X(AIRSteam_SetItemPreview) /* = 61 */
X(AIRSteam_AddItemKeyValueTag) /* = 62 */
X(AIRSteam_RemoveItemKeyValueTags) /* = 63 */
X(AIRSteam_SubmitItemUpdate) /* = 64 */
X(AIRSteam_SubscribeItem) /* = 65 */
X(AIRSteam_UnsubscribeItem) /* = 66 */
X(AIRSteam_GetSubscribedItems) /* = 67 */
X(AIRSteam_GetItemState) /* = 68 */
X(AIRSteam_GetItemInstallInfo) /* = 69 */
X(AIRSteam_GetItemDownloadInfo) /* = 70 */
X(AIRSteam_DownloadItem) /* = 71 */
/***********/
/* Friends */
/***********/
X(AIRSteam_GetFriendCount) /* = 72 */
X(AIRSteam_GetFriendByIndex) /* = 73 */
X(AIRSteam_GetFriendPersonaName) /* = 74 */
/******************************/
/* Authentication & Ownership */
/******************************/
X(AIRSteam_GetAuthSessionTicket) /* = 75 */
X(AIRSteam_GetAuthSessionTicketResult) /* = 76 */
X(AIRSteam_BeginAuthSession) /* = 77 */
X(AIRSteam_EndAuthSession) /* = 78 */
X(AIRSteam_CancelAuthTicket) /* = 79 */
X(AIRSteam_UserHasLicenseForApp) /* = 80 */
/***********/
/* Overlay */
/***********/
X(AIRSteam_ActivateGameOverlay) /* = 81 */
X(AIRSteam_ActivateGameOverlayToUser) /* = 82 */
X(AIRSteam_ActivateGameOverlayToWebPage) /* = 83 */
X(AIRSteam_ActivateGameOverlayToStore) /* = 84 */
X(AIRSteam_ActivateGameOverlayInviteDialog) /* = 85 */
X(AIRSteam_IsOverlayEnabled) /* = 86 */
X(AIRSteam_SetOverlayNotificationPosition) /* = 87 */
/***********************/
/* DLC / subscriptions */
/***********************/
X(AIRSteam_IsSubscribedApp) /* = 88 */
X(AIRSteam_IsDLCInstalled) /* = 89 */
X(AIRSteam_GetDLCCount) /* = 90 */
X(AIRSteam_InstallDLC) /* = 91 */
X(AIRSteam_UninstallDLC) /* = 92 */
X(AIRSteam_DLCInstalledResult) /* = 93 */
/********************/
/* Microtransaction */
/********************/
X(AIRSteam_MicroTxnResult) /* = 94 */
/**********************************/
/* Other non-Steamworks functions */
/**********************************/
X(AIRSteam_GetEnv) /* = 95 */
X(AIRSteam_SetEnv) /* = 96 */
/* END GENERATED CODE */
